sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/utils/JsonUtils.java

```
@@ -18,42 +18,87 @@
-      .registerModule(new JavaTimeModule());
+  private static final String SINGLE_QUOTE = "'";
+  private static final String DOUBLE_QUOTE = "\"";
+
+  protected static final ObjectMapper mapper = mapper();
+
+  public static ObjectMapper mapper() {
+    val mapper = new ObjectMapper().registerModule(new ParameterNamesModule())
+        .registerModule(new Jdk8Module())
+        .registerModule(new JavaTimeModule());
+    mapper.disable(DeserializationFeature.FAIL_ON_MISSING_CREATOR_PROPERTIES);
+    mapper.disable(DeserializationFeature.FAIL_ON_UNRESOLVED_OBJECT_IDS);
+    mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+    mapper.disable(DeserializationFeature.FAIL_ON_INVALID_SUBTYPE);
+    mapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
+
+    mapper.enable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS);
+    return mapper;
+  }
 
   public static JsonNode readTree(String json) throws JsonProcessingException, IOException {
     return mapper.readTree(json);
   }
 
+  public static ObjectNode ObjectNode() {
+    return mapper.createObjectNode();
+  }
+
   @SneakyThrows
   public static String nodeToJSON(ObjectNode node) {
     return mapper.writeValueAsString(node);
   }
 
   @SneakyThrows
-  public static String nodeToJSON(JsonNode node) {
-    return mapper.writeValueAsString(node);
+  public static String toJson(Object o) {
+    return mapper.writeValueAsString(o);
   }
 
-  public static ObjectMapper mapper() {
-    return mapper;
+  @SneakyThrows
+  public static <T> T fromJson(String json, Class<T> toValue) {
+    return mapper.convertValue(mapper.readTree(json), toValue);
+  }
+
+  @SuppressWarnings("unchecked")
+  public static Map<String, Object> toMap(String json, String keyName) {
+    if (json == null || json.equals("")) {
+      json = "{}";
+    }
+    try {
+      return mapper.convertValue(mapper.readTree(json), Map.class);
+    } catch (IllegalArgumentException | IOException e) {
```

Interesting point about the XML. Keep `JsonUtils` clean and fail fast and move the exception catching and handling into the ` Metadata` class.